controllers/openstackcluster_controller.go

```diff
@@ -184,7 +184,7 @@ func (r *OpenStackClusterReconciler) reconcileDelete(ctx context.Context, scope
 	}
 
 	// if ManagedSubnets was not set, no network was created.
-	if len(openStackCluster.Spec.ManagedSubnets) == 0 {
+	if len(openStackCluster.Spec.ManagedSubnets) > 0 {
```

Comment:
https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1856/files#diff-39b99bc838c9f76967790a10572cc44f0c41ccacddcd6e1eff934828d4b050a3L186-R187

Before we were doing:
```golang
if openStackCluster.Spec.NodeCIDR != "" {
```

So it makes sense to check for the length of `ManagedSubnets`.